libethash/data_sizes.h

```diff
@@ -33,213 +31,780 @@ extern "C" {
-        38928741248U, 39041738368U, 39154739584U, 39267741824U, 39380739712U,
-        39493735808U, 39606741632U, 39719741312U, 39832741504U, 39945739648U,
-        40058740352U, 40171740032U, 40284740992U, 40397740672U, 40510740352U,
-        40623740288U, 40736738176U, 40849737856U, 40962741376U, 41075739776U,
-        41188737664U, 41301735808U, 41414738048U, 41527741312U, 41640740992U,
-        41753739904U, 41866739072U, 41979738496U, 42092740736U, 42205739648U,
-        42318740608U, 42431741312U, 42544738688U, 42657741184U, 42770738048U,
-        42883741568U, 42996741248U, 43109740928U, 43222736512U, 43335741056U,
-        43448730496U, 43561740416U, 43674741632U, 43787740544U, 43900741504U,
-        44013739648U, 44126740864U, 44239740544U, 44352741248U, 44465738368U,
-        44578735232U, 44691739264U, 44804741504U, 44917741696U, 45030741376U,
-        45143741824U, 45256740992U, 45369739136U, 45482740096U, 45595739776U,
-        45708739712U, 45821740672U, 45934741376U, 46047741056U, 46160741248U,
-        46273737088U, 46386740864U, 46499739008U, 46612739968U, 46725735296U,
-        46838740864U, 46951741568U, 47064737152U, 47177741696U, 47290741376U,
-        47403738752U, 47516741248U, 47629739648U, 47742741632U, 47855737984U,
-        47968740224U, 48081738368U, 48194741632U, 48307739264U, 48420739712U,
-        48533739136U, 48646738304U, 48759741824U, 48872741504U, 48985739392U,
-        49098741376U, 49211741056U, 49324740992U, 49437738368U, 49550740864U,
-        49663735424U, 49776737408U, 49889740672U, 50002738816U, 50115738752U,
-        50228739712U, 50341741696U, 50454736768U, 50567738752U, 50680739968U,
-        50793736832U, 50906734976U, 51019741568U, 51132739456U, 51245741696U,
-        51358741376U, 51471741056U, 51584738944U, 51697734272U, 51810739072U,
-        51923736448U, 52036740736U, 52149741184U, 52262737024U, 52375738496U,
-        52488740992U, 52601739136U, 52714740352U, 52827736448U, 52940738176U,
-        53053741696U, 53166740864U, 53279741824U, 53392741504U, 53505739136U,
-        53618739584U, 53731741312U, 53844741248U, 53957741696U, 54070741376U,
-        54183740288U, 54296741504U, 54409741696U, 54522739072U, 54635737472U,
-        54748741504U, 54861736064U, 54974740096U, 55087741568U, 55200733568U,
-        55313741696U, 55426734464U, 55539741056U, 55652741504U, 55765741184U,
-        55878741376U, 55991730304U, 56104740992U, 56217740672U, 56330731648U,
-        56443737472U, 56556724352U, 56669740672U, 56782739072U, 56895740032U,
-        57008741248U, 57121741696U, 57234740096U, 57347741312U, 57460741504U
+//       Reap[
+//         While[j < n,
+//           Module[{i =
+//             Floor[(CacheSizeBytesInit + CacheGrowth * j) / HashBytes]},
+//             While[! PrimeQ[i], i--];
+//             Sow[i*HashBytes]; j++]]]][[2]][[1]]
+
+
+static const size_t dag_sizes[2048] = {
+        1073739904U, 1082130304U, 1090514816U, 1098906752U, 1107293056U,
```

Comment:
hard-coding this is utterly horrible. prefer to provide the function to calculate it alongside the data.